### PR TITLE
CORDA-787 Remove unused class

### PR DESCRIPTION
**Pull request by andr3ej**
and minor refactorings

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/ArtemisMessagingComponent.kt [hunk 1]
```diff
@@ -66,7 +65,4 @@ abstract class ArtemisMessagingComponent : SingletonSerializeAsToken() {
     data class ServiceAddress(val identity: PublicKey) : ArtemisAddress, MessageRecipientGroup {
         override val queueName: String = "$PEERS_PREFIX${identity.toBase58String()}"
     }
-
-    /** The config object is used to pass in the passwords for the certificate KeyStore and TrustStore */
-    abstract val config: SSLConfiguration?
```
Comment (andr3ej) [inline node-api/src/main/kotlin/net/corda/nodeapi/ArtemisMessagingComponent.kt]: not used here

File: node/src/main/kotlin/net/corda/node/services/api/AbstractNodeService.kt [hunk 2]
```diff
@@ -1,56 +0,0 @@
...
-import net.corda.core.serialization.deserialize
-import net.corda.core.serialization.serialize
-import net.corda.node.services.messaging.*
-import javax.annotation.concurrent.ThreadSafe
-
-/**
- * Abstract superclass for services that a node can host, which provides helper functions.
- */
-@ThreadSafe
-abstract class AbstractNodeService(val network: MessagingService) : SingletonSerializeAsToken() {
```
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/services/api/AbstractNodeService.kt]: not used

File: node/src/main/kotlin/net/corda/node/services/messaging/Messaging.kt [hunk 3]
```diff
@@ -205,19 +205,6 @@ fun MessagingService.send(topic: String, sessionID: Long, payload: Any, to: Mess
...
-/**
- * This class lets you start up a [MessagingService]. Its purpose is to stop you from getting access to the methods
- * on the messaging service interface until you have successfully started up the system. One of these objects should
- * be the only way to obtain a reference to a [MessagingService]. Startup may be a slow process: some implementations
- * may let you cast the returned future to an object that lets you get status info.
- *
- * A specific implementation of the controller class will have extra features that let you customise it before starting
- * it up.
- */
-interface MessagingServiceBuilder<out T : MessagingService> {
```
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/services/messaging/Messaging.kt]: only used in one file, into which i've moved it

File: node/src/test/kotlin/net/corda/node/services/messaging/ArtemisMessagingTests.kt [hunk 4]
```diff
@@ -215,7 +211,7 @@ class ArtemisMessagingTests {
                     identity.public,
                     ServiceAffinityExecutor("ArtemisMessagingTests", 1),
                     database,
-                    MonitoringService(MetricRegistry())).apply {
```
Comment (andr3ej) [inline node/src/test/kotlin/net/corda/node/services/messaging/ArtemisMessagingTests.kt]: one fewer thing to "mock"

File: node-api/src/main/kotlin/net/corda/nodeapi/ArtemisMessagingComponent.kt [hunk 5]
```diff
@@ -4,16 +4,14 @@ import net.corda.core.messaging.MessageRecipientGroup
...
-import net.corda.core.serialization.SingletonSerializeAsToken
 import net.corda.core.utilities.NetworkHostAndPort
 import net.corda.core.utilities.toBase58String
-import net.corda.nodeapi.config.SSLConfiguration
 import java.security.PublicKey
 
 /**
  * The base class for Artemis services that defines shared data structures and SSL transport configuration.
  */
-abstract class ArtemisMessagingComponent : SingletonSerializeAsToken() {
```
Comment (mnesbit) [inline node-api/src/main/kotlin/net/corda/nodeapi/ArtemisMessagingComponent.kt]: I agree with the change, but I think that technically this is API change.

Comment (andr3ej) [reply node-api/src/main/kotlin/net/corda/nodeapi/ArtemisMessagingComponent.kt]: in what way?

Comment (mnesbit) [reply node-api/src/main/kotlin/net/corda/nodeapi/ArtemisMessagingComponent.kt]: Only in the technical sense that this class is in the node-api library, not marked as internal.

Comment (andr3ej) [reply node-api/src/main/kotlin/net/corda/nodeapi/ArtemisMessagingComponent.kt]: moved to internal

**Pull request merged**